main.go

```diff
@@ -0,0 +1,226 @@
+		if err != nil {
+			return err
+		}
+
+		if isSymlink {
+			path = evaledPath
+		}
+
+		info, err = os.Lstat(path)
+		if err != nil {
+			return err
+		}
+
+		header, err := zip.FileInfoHeader(info)
+		if err != nil {
+			return err
+		}
+
+		if baseDir != "" {
+			if isSymlink {
+				header.Name = filepath.Join(baseDir, strings.TrimPrefix(originalPath, sourcePath))
+			} else {
+				header.Name = filepath.Join(baseDir, strings.TrimPrefix(path, sourcePath))
+			}
+
+		} else {
+			baseDir = "/"
+		}
+
+		if info.IsDir() {
+			header.Name += "/"
+		} else {
+			header.Method = zip.Deflate
+		}
+
+		writer, err := archive.CreateHeader(header)
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		file, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		_, err = io.Copy(writer, file)
+		return err
+	})
+
+	return fmt.Sprint(zipfile), nil
+}
+
+// If the file is a symbolic link it will evaulate the path name.
+
+// If the target is a symbolic link it will return true, the evaulated path, and the original path.
+// If the target is  not a symbolic link it will return false, and the original path.
+// If there is an error it will return the error.
+func checkSymlink(path string) (isSymlink bool, evaledPath string, originalPath string, error error) {
+	info, err := os.Lstat(path)
+	if err != nil {
+		return false, "", path, err
+	}
+
+	if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+		evaledPath, err := filepath.EvalSymlinks(path)
+		if err != nil {
+			return true, "", path, err
+		}
+
+		return true, evaledPath, path, nil
+	}
+	return false, "", path, nil
+}
+
+// It will check the target path's existence.
+// If the targetPath does not exist it will create that.
+func validateTargetPath(targetPath string) {
+	dirOftargetPath := filepath.Dir(targetPath)
+	fmt.Print(dirOftargetPath)
+	if err := input.ValidateIfPathExists(dirOftargetPath); err != nil {
+		log.Printf("targetRootPath: %s does not exist", dirOftargetPath)
+		os.MkdirAll(dirOftargetPath, os.ModePerm)
+	}
+}
+
+func createConfigsModelFromEnvs() ConfigsModel {
+
+	return ConfigsModel{
+		SourcePath: os.Getenv("source_path"),
+		TargetPath: os.Getenv("target_path"),
+	}
+}
+
+func (configs ConfigsModel) print() {
+	log.Infof("Create ZIP configs:")
+	log.Printf("- SourcePath: %s", configs.SourcePath)
+	log.Printf("- TargetPath: %s", configs.TargetPath)
+}
+
+func (configs ConfigsModel) validate() error {
+	if err := input.ValidateIfNotEmpty(configs.SourcePath); err != nil {
+		return errors.New("issue with input SourcePath: " + err.Error())
+	}
+
+	if err := input.ValidateIfNotEmpty(configs.TargetPath); err != nil {
+		return errors.New("issue with input TargetPath: " + err.Error())
+	}
+
+	if err := input.ValidateIfDirExists(configs.SourcePath); err != nil {
+		return errors.New("issue with input SourcePath: " + err.Error())
+	}
+
+	checkAlreadyExist(configs)
```

Comment:
do not do this here. later you create the current full target path, validate it there